wasm/src/main/java/com/dylibso/chicory/wasm/types/MemoryLimits.java

```diff
@@ -63,14 +65,12 @@ public MemoryLimits(int initial, int maximum) {
      * @param shared {@code true} if the limits apply to a shared memory segment, or {@code false} otherwise
      */
     public MemoryLimits(int initial, int maximum, boolean shared) {
-        if (initial < 0 || initial > maximum) {
-            throw new IllegalArgumentException(
-                    "initial must be >= 0 and <= maximum, but was " + initial);
+        if (initial > MAX_PAGES || maximum > MAX_PAGES || initial < 0 || maximum < 0) {
```

Comment:
Could do a separate check for negative, unless this error message is required per specification.
```java
if (initial < 0 || maximum < 0) {
    throw new InvalidException("memory size is negative");
} else if (initial > MAX_PAGES || maximum > MAX_PAGES) {

Reply:
> unless this error message is required per specification.

It's the case, for context `< 0` usually happens during overflows other than real negative input passed by the user.